runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobInvocation.java

```diff
@@ -71,58 +48,17 @@ private SamzaPipelineResult invokeSamzaJob() {
     }
   }
 
-  @Override
-  public void start() {
```

Comment:
The whole point of this PR is I thought it might be useful to share much of `FlinkJobInvocation`'s logic with other runners. With these changes, Samza runner should be doing the exact same things as before, except inside a future. Is there some reason that wouldn't be desirable?

Reply:
That makes sense!